incubator/crate/templates/statefulset.yaml

```
@@ -0,0 +1,98 @@
+apiVersion: "apps/v1"
+kind: StatefulSet
+metadata:
+  name: crate
+spec:
+  serviceName: "crate-db"
+  replicas: {{ .Values.crate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Values.app }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.app }}
+        app.kubernetes.io/name: crate-deployment
+        app.kubernetes.io/part-of: crate-db
+        app.kubernetes.io/version: {{ .Values.image.tag }}
+        app.kubernetes.io/managed-by: helm
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      initContainers:
+      - name: init-sysctl
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+        securityContext:
+          privileged: true
+      containers:
+      - name: crate
+        image: {{ .Values.image.name }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+          - /docker-entrypoint.sh
+          - -Clicense.enterprise=false
+          - -Ccluster.name={{ .Values.crate.clusterName }}
+          - -Cnetwork.host="0.0.0.0"
+          - -Cdiscover.seed_providers=srv
```

The configuration is missing the `cluster.initial_master_nodes` setting.

```suggestion
          - -Cdiscover.seed_providers=srv
          - -Cnode.name=#{POD_NAME}
          - -Ccluster.initial_master_nodes={{ range $i, $sequence :=  (seq .Values.crate.replicas) }}crate-{{ $i }},{{ end}}
```

`POD_NAME` is obtains from the metadata.name, see comment below.

PS: I am not familiar with the Go templating, so the loop may not be accurate. It should output the names of the pod, separated by comma, e.g.  if there are 3 replicas in this Statefulset, the line should read like so:

```yaml
          - -Ccluster.initial_master_nodes=crate-0,crate-1,crate-2
```

Where `crate` is the name of the Pod (defined by `metadata.name`.